pkg/monitor/agent/listener1_2.go

```diff
@@ -78,3 +81,11 @@ func (ml *listenerv1_2) drainQueue() {
 func (ml *listenerv1_2) Version() listener.Version {
 	return listener.Version1_2
 }
+
+// Close closes the underlying socket and .
+func (ml *listenerv1_2) Close() {
+	ml.once.Do(func() {
+		close(ml.queue)
```

Comment:
Interesting, were we leaking the ml.queue channel here before? I wonder if we need a separate bugfix for older releases.

Reply:
yes it does look like ml.queue was never closed. i can split this part into a separate PR if it makes it easier to backport.

Reply:
@michi-covalent please do. Looks like we'll wait until branching for the next release before we merge this PR anyway.

Reply:
on the second thought, looks like go garbage collects unused channels? https://groups.google.com/forum/#!msg/golang-nuts/pZwdYRGxCIk/qpbHxRRPJdUJ

i needed to add `close(ml.queue)` in this pr to unblock `drainQueue` from here: https://github.com/cilium/cilium/blob/c6e876c1ac568db853af21ccde5445cef9da1a5e/pkg/monitor/agent/listener1_2.go#L63

Reply:
OK, in that case don't worry.